packages/salesforcedx-vscode-apex/src/language-server.ts

```diff
@@ -20,83 +20,86 @@ import { nls } from './messages';
 import * as requirements from './requirements';
 
 const UBER_JAR_NAME = 'apex-jorje-lsp.jar';
-const JDWP_DEBUG_PORT = 2739;
 const APEX_LANGUAGE_SERVER_MAIN = 'apex.jorje.lsp.ApexLanguageServerLauncher';
 
 declare var v8debug: any;
 const DEBUG = typeof v8debug === 'object' || startedInDebugMode();
 
+function getFreePort(): Promise<number> {
+  return new Promise((resolve, reject) => {
+    portFinder.getPort((err, port) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(port);
+      }
+    });
+  });
+}
+
 async function createServer(
   context: vscode.ExtensionContext
 ): Promise<StreamInfo> {
   try {
     const requirementsData = await requirements.resolveRequirements();
-    return new Promise<any>((resolve, reject) => {
-      portFinder.getPort((err, port) => {
-        const uberJar = path.resolve(
-          context.extensionPath,
-          'out',
-          UBER_JAR_NAME
-        );
-        const javaExecutable = path.resolve(
-          `${requirementsData.java_home}/bin/java`
-        );
-        let args: string[];
-        if (DEBUG) {
-          args = [
-            '-cp',
-            uberJar,
-            `-Dapex-lsp.port=${port}`,
-            '-Ddebug.internal.errors=true',
-            '-Ddebug.semantic.errors=false',
-            '-Dtrace.protocol=false',
-            `-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=${JDWP_DEBUG_PORT}`,
-            APEX_LANGUAGE_SERVER_MAIN
-          ];
-        } else {
-          args = [
-            '-cp',
-            uberJar,
-            `-Dapex-lsp.port=${port}`,
-            '-Ddebug.internal.errors=true',
-            '-Ddebug.semantic.errors=false',
-            APEX_LANGUAGE_SERVER_MAIN
-          ];
-        }
+    return new Promise<any>(async (resolve, reject) => {
+      const port = await getFreePort();
+      const uberJar = path.resolve(context.extensionPath, 'out', UBER_JAR_NAME);
+      const javaExecutable = path.resolve(
+        `${requirementsData.java_home}/bin/java`
+      );
+      let args: string[];
+      if (DEBUG) {
+        args = [
+          '-cp',
+          uberJar,
+          `-Dapex-lsp.port=${port}`,
+          '-Ddebug.internal.errors=true',
+          '-Ddebug.semantic.errors=false',
+          '-Dtrace.protocol=false',
+          `-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=${port +
```

Comment:
I am aware of this limitation (that the port + 1) is not available.
Since this is for local debugging purposes, I am willing to tolerate that for now.

I tried calling `getFreePort()` twice in this session but it only returns the same port. So, I must be misunderstanding its interface (or guarantees).